pkg/syncer/syncer.go

```diff
@@ -316,12 +390,28 @@ func (c *Controller) process(gvr schema.GroupVersionResource, obj interface{}) e
 			return err
 		}
 	}
-	namespace, name := meta.GetNamespace(), meta.GetName()
-	if c.inSyncerNamespace(namespace) {
+
+	namespace := meta.GetNamespace()
+
+	// This must be checked before the namespace is transformed
+	if c.inSyncerNamespace(meta) {
 		klog.V(2).Infof("Skipping object of type: %T : %v in syncer namespace", obj, obj)
 		return nil
 	}
 
+	var targetName, targetNamespace string
+	var err error
+	if c.direction == KcpToPcluster {
+		targetName, targetNamespace, err = transformForCluster(gvr, meta)
+	} else {
+		targetName, targetNamespace, err = transformForKcp(gvr, meta)
+	}
+
+	if err != nil {
+		klog.Error(err)
+		return err
```

Comment:
If you return an error here, the controller logic will requeue the same key (with backoff). Presumably we'd encounter the same error again. It's probably better to return nil here to avoid that.

Reply:
+1, if you do that, consider logging the error with an indication that we're dropping as a result. It's tough to tell from logs sometimes whether an error means "an error! (don't worry, I'll retry)" vs "an error! (you'll never see this again)"